src/main/webapp/user-location.js

```diff
@@ -0,0 +1,60 @@
+const apiKey = 'AIzaSyB5cba6r-suEYL-0E_nRQfXDtT4XW0WxbQ';
+const fetchJson = {
+  'considerIp': 'true',
+};
+const geoCodingUrl = 'https://maps.googleapis.com/maps/api/geocode/json?latlng=';
+const geoLocationUrl = 'https://www.googleapis.com/geolocation/v1/geolocate?key=';
+
+// Given the user's lat and lng from the geoLocation API,
+// getUserState calls to the geoCoding API to reverse geoCode
+// the coordinates and get the user's current state.
+async function getUserState(lat, lng) {
+  const geoCodingFetch = geoCodingUrl + lat + ',' + lng + '&key=' + apiKey;
+  return fetch(geoCodingFetch).then(function(response) {
+      return response.json();
+    })
+    .then(function(jsonResponse) {
+      // Traverse all of the addresses and find one that
+      // has administrative_area_level_1 (state) as one of the address
+      // components.Then return the addressComponent.long_name
+      // which will be the state name.
+      const results = jsonResponse.results;
+      let state;
+      results.forEach((address) => {
+        const addressComponents = address.address_components;
+        addressComponents.forEach((component) => {
+          const types = component.types;
+          types.forEach((type) => {
+            if (type === 'administrative_area_level_1') {
+              state = component.long_name;
```

Comment:
Is it ever possible that we could fail to find this and then be returning nothing for state? Maybe you could add an error check to getDefaultValue where if the state returned is undefined or empty, or an error is thrown at any point during the function, you just return 'each U.S. state' instead.

Reply:
I haven't entered in a ton of points to test the API's accuracy, but I'm sure it is possible. Will add error checking now.

Reply:
Fixed!